src/operations/operation.ts

```diff
@@ -49,21 +47,22 @@ export abstract class OperationBase<
   // BSON serialization options
   bsonOptions?: BSONSerializeOptions;
 
-  constructor(options: T = {} as T) {
-    this.options = Object.assign({}, options);
+  [kSession]: ClientSession;
 
+  constructor(options: OperationOptions = {}) {
     this.readPreference = this.hasAspect(Aspect.WRITE_OPERATION)
       ? ReadPreference.primary
       : ReadPreference.fromOptions(options) ?? ReadPreference.primary;
-    // TODO: A lot of our code depends on having the read preference in the options. This should
-    //       go away, but also requires massive test rewrites.
```

Comment:
I guess this didn't end up breaking that many tests, I'm seeing no more than 30 at the time of writing, does that seem correct?

Reply:
This piece of code was particularly nasty, and lead to a lot of unexpected failures. Ultimately, removing it makes the code easier to reason about, but it required moving code from `session.ts` to `executeOperation`